SharedPackages/BrowserServicesKit/Sources/Networking/Auth/OAuthClient.swift

```diff
@@ -202,20 +200,21 @@ final public actor DefaultOAuthClient: @preconcurrency OAuthClient {
     // MARK: - Public
 
     public var isUserAuthenticated: Bool {
-        tokenStorage.tokenContainer != nil
+        let tokenContainer = try? tokenStorage.getTokenContainer()
```

Comment:
This is... interesting.

You're not changing anything that wasn't already here.  This isn't new and I don't want to add scope creep.

That said: do you think we should follow up on this, after this PR?  Should this throw instead of returning `false`?

Reply:
here is where we hit issues with backward compatibility with AuthV1,  isUserAuthenticated is defined in the protocol `SubscriptionAuthenticationStateProvider` that is implemented by both AuthV1 and V2, changing this would mean changing authv1 and that would cause a couple of issues:
 1. Authv1 error handling is not good enough for propagating an error thrown here, the change would require a, difficult to estimate, chain of changes.
 2. We don't really want to refactor authv1 at this stage because we'll hopefully be able to remove it altogether very soon and we want to minimize risks.

Reply:
Yeah, again - I'm not looking to push changes here, just wondering if there might be issues in relation to this.

I'm content with knowing it's in your radar.